src/azure-cli/azure/cli/command_modules/acr/_utils.py

```diff
@@ -431,7 +437,36 @@ def parse_actions_from_repositories(allow_or_remove_repository):
 
     return actions
 
+def prepare_source_location(source_location, client_registries, registry_name, resource_group_name):
+    if source_location is None or source_location.lower() == ACR_NULL_CONTEXT:
+        source_location = None
+    elif os.path.exists(source_location):
+        if not os.path.isdir(source_location):
+            raise CLIError(
+                "Source location should be a local directory path or remote URL.")
+
+        tar_file_path = os.path.join(tempfile.gettempdir(
+        ), 'cli_source_archive_{}.tar.gz'.format(uuid.uuid4().hex))
+
+        try:
+            source_location = upload_source_code(
+                client_registries, registry_name, resource_group_name,
+                source_location, tar_file_path, "", "")
+        except Exception as err:
+            raise CLIError(err)
+        finally:
+            try:
+                logger.debug(
+                    "Deleting the archived source code from '%s'...", tar_file_path)
+                os.remove(tar_file_path)
+            except OSError:
+                pass
+    else:
+        source_location = check_remote_source_code(source_location)
```

Comment:
I am assuming this is the logic that currently handles the empty string `source_location`